test/settings.py

```diff
@@ -1,11 +1,28 @@
 """Constants used in test files."""
 
-HOST = "127.0.0.1"
-PORT = "5050"
-L1_HOST = "localhost"
-L1_PORT = "8545"
-
-APP_URL = f"http://{HOST}:{PORT}"
-GATEWAY_URL = APP_URL
-FEEDER_GATEWAY_URL = APP_URL
-L1_URL = f"http://{L1_HOST}:{L1_PORT}/"
+import socket
+
+# pylint: disable=invalid-name,too-few-public-methods
+class Settings:
```

Comment:
Any advantage why it's a class that gets instantiated and not a class with a class method? Also, why even convert this into a class and not keep it as root-level constants?

Reply:
Do you mean like the static method? Because then it cant change self and I need like 'singleton'.
Because as root level dev can import var ex. PORT in test and in runtime port will be changed to designated port but the imported PORT is not by reference so it will stay unchanged. So to avoid that originally I created get_port() and set_port() functions that can then be used to fix reference problems as they will be exec in runtime and get new value but still you could make mistake using the imported PORT for ex.
Then I try to create something like JS getter and setter for Vars but can't find python way to make it as in JS so the class was the final solution to make it clean and unable for dev to make mistake.

Reply:
Three approaches here:
1) You can use a `@staticmethod` which access class properties directly by class name, not through `self`
2) You can use a `@classmethod` which has access to `cls` instead of `self`.
3) As suggested here: https://github.com/Shard-Labs/starknet-devnet/pull/165#discussion_r915755873, the port can be assigned directly if used with the helper function, so no need for accessing properties after their definition.